src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -58,9 +60,35 @@ public function getSenders(Envelope $envelope): iterable
                     $seen[] = $senderAlias;
 
                     yield from $this->getSenderFromAlias($senderAlias);
+                    $found = true;
                 }
             }
         }
+
+        // Let the configuration-driven map upper override message attributes,
+        // this allows environment-specific configuration overriding hardcoded
+        // transport name.
+        if (!$found) {
+            foreach ($this->getTransportNamesFromAttribute($envelope) as $senderAlias) {
+                yield from $this->getSenderFromAlias($senderAlias);
+            }
+        }
+    }
+
+    private function getTransportNamesFromAttribute(Envelope $envelope): array
+    {
+        $transports = [];
+        $message = $envelope->getMessage();
+
+        foreach ((new \ReflectionClass($message))->getAttributes(AsMessage::class) as $refAttr) {
```

Comment:
either we need the instanceof flag, or we should make the attribute final

Reply:
I’d use the instanceof flag here, as you might want to create your own attribute with a common transport for several messages

Reply:
You mean iterating over all attributes without filter and check using instanceof ? If so OK, but I'd rather use is_subclass_of() instead of creating the new instance for each ?

Reply:
Ah OK forget everything I said, I wasn't aware of `ArgumentMetadata::IS_INSTANCEOF`.

Reply:
OK I wasn't aware of the `ReflectionAttribute::IS_INSTANCEOF` flag, fine by me, thanks for making me discover this!